src/Tickets/Emails/Email_Template.php

```diff
@@ -0,0 +1,193 @@
+<?php
+/**
+ * Handles registering and setup for the Tickets Emails settings tab.
+ *
+ * @since TBD
+ *
+ * @package TEC\Tickets\Emails
+ */
+
+namespace TEC\Tickets\Emails;
+
+use Tribe__Template;
+use Tribe__Tickets__Main;
+
+/**
+ * Class Email_Template
+ *
+ * @since   TBD
+ *
+ * @package TEC\Tickets\Emails
+ */
+class Email_Template {
+
+	private bool $preview = false;
+
+	private Array $context_data = [];
+	
+	/**
+	 * Gets the template instance used to setup the rendering html.
+	 *
+	 * @since TBD
+	 *
+	 * @return Tribe__Template The template object.
+	 */
+	public function get_template(): Tribe__Template {
+		if ( empty( $this->template ) ) {
+			$this->template = new Tribe__Template();
+			$this->template->set_template_origin( Tribe__Tickets__Main::instance() );
+			$this->template->set_template_folder( 'src/admin-views/settings/emails' );
+			$this->template->set_template_context_extract( true );
+		}
+
+		return $this->template;
+	}
+
+	/**
+	 * Returns the email template HTML.
+	 *
+	 * @return string The HTML of the template.
+	 */
+	public function get_html() {
+		$template = $this->get_template();
+		return $template->template( 'email-template', $this->get_context(), false );
+	}
+
+	/**
+	 * Prints the email template HTML.
+	 *
+	 * @return void.
+	 */
+	public function render() {
+		echo $this->get_html();
+	}
+
+	/**
+	 * Sets whether or not this will be a template preview.
+	 *
+	 * @since TBD
+	 *
+	 * @param boolean $is_preview
+	 * 
+	 * @return void
+	 */
+	public function set_preview( $is_preview = false ) {
+		$this->preview = $is_preview;
+	}
+
+	/**
+	 * Is this a template preview?
+	 *
+	 * @since TBD
+	 *
+	 * @return boolean Whether or not this is a template preview.
+	 */
+	public function is_preview() {
```

Comment:
Let's add the return type.

Reply:
Am I documenting it wrong? I have `@return boolean...` in the DocBlock.

Reply:
Added more context here https://github.com/the-events-calendar/event-tickets/pull/2518#discussion_r1047304878